pkg/controller/compliancescan/scantype.go

```diff
@@ -145,21 +146,33 @@ func (nh *nodeScanTypeHandler) createScanWorkload() error {
 	return nil
 }
 
-func (nh *nodeScanTypeHandler) handleRunningScan() (bool, error) {
+func (nh *nodeScanTypeHandler) handleRunningScan() (bool, []string, error) {
+	// scan.Spec.ComplianceScanSettings.Timeout is in string format, e.g. "1h30m"
+	// so we need to parse it
+	timeoutVal := podTimeoutDisable
+	timeoutNodes := []string{}
+	var err error
+	if nh.scan.Spec.ComplianceScanSettings.Timeout != "" {
+		timeoutVal, err = time.ParseDuration(nh.scan.Spec.ComplianceScanSettings.Timeout)
```

Comment:
I wonder if this should be moved to `ReconcileComplianceScan.validate()`

Reply:
The timeout value is being passed to isPodRunningInNode, wondering what should I do here?

Reply:
(sorry for the late reply)
I would say parse twice, once in `validate` and once before use. The reason is that if the duration fails parsing on validate we would go straight to `DONE/ERROR` whereas if the parsing fails just before use, the reconcile loop would retry indefinitely.

Reply:
@Vincent056 all my comments about the PR have been addressed except for this one - let me know if you disagree and I should just LGTM the patch. Otherwise great work!

Reply:
Sorry for the delay, it has been fixed now!